### PR TITLE
site: Add search

### PR DESCRIPTION
**Pull request by askoufis**
[**Preview site**](https://63fc30f6771c470d7338fad6--vanilla-extract-css.netlify.app)

Created a `SearchInput` component that wraps algolia's `DocSearch` component.

Not customizing too much for now, the two notable customizations are:
- Making search result URLs relative rather than absolute so clicking a search result when running the site locally/on a preview site doesn't take you to production
- Customizing the missing result URL to point to a github issue template

[DocSearch component API Reference](https://docsearch.algolia.com/docs/api)

I've also taken a crack at the styling and general positioning on site. I've put it next to the color mode icon/fab, and attempted to style it in the same manor as the fab, mainly for consistency. Feel free to clean up or change the styles completely, I don't have any strong opinions about them. The docesarch styles can be found [here](https://github.com/algolia/docsearch/tree/main/packages/docsearch-css/src), and I've overwritten some of the CSS vars and styles with global styles.

**NOTE**: There are a handful of CSS vars you can customize, but some of them are used for _both_ the modal and the search input, which is a bit of a pain.

### CODE REVIEW

File: site/src/Fab/Fab.css.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export const fab = style({
   height: fabSize,
   width: fabSize,
   zIndex: 3,
-  boxShadow: `0px 0px 0px 5px ${fallbackVar(focusColorVar, 'transparent')}`,
+  boxShadow: `0px 0px 0px 3px ${fallbackVar(focusColorVar, 'transparent')}`,
```
Comment (askoufis) [inline site/src/Fab/Fab.css.ts:13]: This change aligns the focus box shadow with the color mode toggle's box shadow.

**Comment by mattcompiles** [comment]
@askoufis This is super cool!

Are we able to tie the heading specific content to their respective hashlinks? e.g. `/documentation/global-api/create-global-theme-contract/#formatting-the-variable-names`

**Comment by askoufis** [comment]
> @askoufis This is super cool!
> 
> Are we able to tie the heading specific content to their respective hashlinks? e.g. `/documentation/global-api/create-global-theme-contract/#formatting-the-variable-names`

Hmm, I feel like this might be an issue with the indexer. I'll seem if there's some settings that need tweaking.

**Comment by askoufis** [comment]
> @askoufis This is super cool!
> 
> Are we able to tie the heading specific content to their respective hashlinks? e.g. `/documentation/global-api/create-global-theme-contract/#formatting-the-variable-names`

I think the issue here is that [the indexer is looking for anchors on heading tags](https://docsearch.algolia.com/docs/tips#add-anchors-to-headings), but the docs site has ids on `a` tags.

**Comment by michaeltaranto** [comment]
Opened the [PR separately], hopefully that can seed the index and we can validate against this PR before merging.

[PR separately]: https://docsearch.algolia.com/docs/tips/#add-anchors-to-headings

**Comment by michaeltaranto** [comment]
Trying out moving the `id` from the anchor to the heading https://github.com/vanilla-extract-css/vanilla-extract/pull/1016

**Comment by askoufis** [comment]
> Trying out moving the `id` from the anchor to the heading #1016

This change worked, anchors are now being correctly identified in the search index. Had to [tweak the search result URL](https://github.com/vanilla-extract-css/vanilla-extract/pull/1006/commits/2c8f987d0ead485e4a7b6da48083e0a01f8076d0) to actually _use_ the anchors, but searching results work correctly now 🎉

**Review by michaeltaranto — APPROVED** [review]
This is feeling good! Happy from my side

File: site/src/themeUtils.ts [hunk 2]
```diff
@@ -1,24 +1,25 @@
...
 import isEqual from 'lodash/isEqual';
 import { StyleRule } from '@vanilla-extract/css';
-import { Properties, SimplePseudos } from 'csstype';
+import { Properties } from 'csstype';
 import mapValues from 'lodash/mapValues';
 
 export const breakpoints = {
   mobile: 0,
-  tablet: 768,
+  tablet: 769,
```
Comment (michaeltaranto) [inline site/src/themeUtils.ts:9]:
```suggestion
  tablet: 769, // aligning breakpoint with the SearchInput, which uses a `max-width: 768px`. Ref: https://github.com/algolia/docsearch/blob/d81016b110aa0818231b6e4b7b96d2007d345b05/packages/docsearch-css/src/button.css#L66
```

**Pull request merged**